python/graphstorm/trainer/ep_trainer.py

```diff
@@ -230,7 +229,10 @@ def fit(self, train_loader, num_epochs,
             output = {'best_test_score': self.evaluator.best_test_score,
                        'best_val_score': self.evaluator.best_val_score,
                        'peak_mem_alloc_MB': th.cuda.max_memory_allocated(device) / 1024 / 1024,
-                       'best_epoch': best_epoch}
+                       'best validation iteration': \
+                           self.evaluator.best_iter_num[self.evaluator.metric[0]],
```

Comment:
You need to check whether no-validation is set.

Reply:
The check has be done by setting up evaluator.

If no-validation is set, there will no evaluator defined. So, this output will be disabled automatically.